### PR TITLE
runtime-rs: Add RNG to QEMU cmdline

### PR DESCRIPTION
**Pull request by emanuellima1**

### CODE REVIEW

**Comment by emanuellima1** [comment]
@pmores @gkurz @beraldoleal please review.
I had a rebasing issue...

**Comment by beraldoleal** [comment]
@emanuellima1 I believe you still have the rebase issue: The PR is showing 5 commits.

You need to rebase on top of main, before pushing the PR. Also your commit message has two lines with `Signed-off-by: Emanuel Lima <emlima@redhat.com>`.

**Comment by emanuellima1** [comment]
Those commits are old and are only there because I started with an older branch. I'm not sure how to fix this :/

File: src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs [hunk 1]
```diff
@@ -1148,6 +1149,61 @@ impl ToQemuParams for Rtc {
...
+impl ToQemuParams for RngDevice {
+    async fn qemu_params(&self) -> Result<Vec<String>> {
+        let mut object_params = Vec::new();
+        let mut device_params = Vec::new();
+
+        object_params.push("rng-random".to_owned());
+        object_params.push(format!("id={}", self.id));
+        object_params.push(format!("filename={} ", self.filename));
+
+        device_params.push(format!("-device {}", self.transport));
```
Comment (beraldoleal) [inline src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs:1193]: isn't `-device` here duplicated as is already added at line 1201? So maybe:

```
device_params.push(self.transport.clone());
```

File: src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs [hunk 2]
```diff
@@ -1148,6 +1149,61 @@ impl ToQemuParams for Rtc {
...
+        device_params.push(format!("rng={}", self.id));
+        device_params.push(format!("max_bytes={}", self.max_bytes));
```
Comment (beraldoleal) [inline src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs:1195]: `s/max_bytes={}/max-bytes={}/g` ?

**Review by beraldoleal — COMMENTED** [review]
Hi @emanuellima1, thanks for this.

I left a few comments for you, but I will leave the proper review to @pmores.

**Comment by beraldoleal** [comment]
> I'm not sure how to fix this :/

Ping me on slack and I can help you.

File: src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs (hunk 1 above)
Comment (emanuellima1) [reply src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs:1193]: Yes, my bad

Comment (emanuellima1) [reply src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs:1193]: Rust is snake_case, so no

**Comment by Apokleos** [comment]
Thx @emanuellima1 I think you should rebase your code first.

**Comment by emanuellima1** [comment]
Please review this [PR](https://github.com/kata-containers/kata-containers/pull/9639) instead

**Pull request closed without merging**